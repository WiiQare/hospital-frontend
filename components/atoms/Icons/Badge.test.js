import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Badge from "./Badge";

describe("Badge", () => {
  let component;
  const total = 10;
  beforeEach(() => {
    const res = render(
      <Badge total={total}>
        <div>test</div>
      </Badge>
    );
    component = res.container;
  });

  it("should render the component", () => {
    expect(component).toMatchSnapshot();
  });

  test("should render text elements", () => {
    expect(screen.getByText(total)).toBeInTheDocument();
  });
});
