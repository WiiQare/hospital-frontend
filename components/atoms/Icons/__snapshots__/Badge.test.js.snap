// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render the component 1`] = `
<div>
  <div
    class="relative inline-flex items-center p-3 text-sm font-medium text-center text-gray-500 hover:text-gray-700"
  >
    <div>
      test
    </div>
    <span
      class="sr-only"
    >
      Notifications
    </span>
    <div
      class="absolute inline-flex items-center justify-center w-6 h-6 text-xs font-bold text-white bg-sky border-2 border-white rounded-full top-0 right-0 dark:border-gray-900"
    >
      10
    </div>
  </div>
</div>
`;
